src/components/App.js

```diff
@@ -0,0 +1,36 @@
+import MainPage from '../pages/MainPage'
+import NotFoundPage from '../pages/NotFoundPage'
+import { DATA, ID } from '../utils/constants'
+import { addEventHandler } from '../domain/domUtils'
+import { fetchMainData } from '../domain/fetchData'
+
+export default function App({ $target }) {
+  let main, notFound
+
+  this.init = () => {
+    addEventHandler(window, 'popstate', this.route)
+    route()
+  }
+
+  const updateState = (targetState, id) => {
+    fetchMainData(main, targetState, id)
+  }
```

Comment:
아래에 fetchMainData함수랑 같이 봐주세요!

```suggestion
  const updateState = async (targetState, id) => {
    const { documents, content } = await fetchMainData(targetState, id)
    documents && main.setDocuments(documents)
    content && main.setContent(content)
  }
```

이런식으로 해도 되고 main.setDocuments 와 main.setContent 에서 빈 값이 온다면 얼리리턴하는 방법도 있겠습니다.

Reply:
무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사무한감사